docs/extensibility/filtering-payment-methods.md

```diff
@@ -0,0 +1,75 @@
+# Filtering payment methods in the Checkout block
+
+## The problem
+
+You're an extension developer, and your extension is conditionally hiding payment gateways on the checkout step. You need to be able to hide payment gateways on the Checkout block using a front-end extensibility point.
+
+## The solution
+
+WooCommerce Blocks provides a function called `registerPaymentMethodExtensionCallbacks` which allows extensions to register callbacks for specific payment methods to determine if they can make payments.
+
+## Importing
+
+_Aliased import_
+
+```js
+import { registerPaymentMethodExtensionCallbacks } from '@woocommerce/blocks-registry';
+```
+
+_wc global_
+
+```js
+const { registerPaymentMethodExtensionCallbacks } = wc.wcBlocksRegistry;
+```
+
+## Signature
+
+| Parameter   | Description                                                                                                         | Type                                              |
+| ----------- | ------------------------------------------------------------------------------------------------------------------- | ------------------------------------------------- |
+| `name`      | Unique string to identify your extension. Choose something that eliminates a name collision with another extension. | `string`                                          |
+| `callbacks` | An object containing callbacks registered for different payment methods                                             | Record< string, CanMakePaymentExtensionCallback > |
+
+Read more below about [callbacks](#callbacks-registered-for-payment-methods).
+
+### Extension name collision
+
+When trying to register callbacks under an extension namespace already used with `registerPaymentMethodExtensionCallbacks`, the registration will be aborted and you will be noticed that you are not using a unique namespace.
```

Comment:
```suggestion
When trying to register callbacks under an extension namespace already used with `registerPaymentMethodExtensionCallbacks`, the registration will be aborted and you will be notified that you are not using a unique namespace. This will be shown in the JavaScript console.
```